crates/core_arch/src/x86/sse.rs

```diff
@@ -1251,6 +1251,19 @@ pub unsafe fn _mm_loadr_ps(p: *const f32) -> __m128 {
     simd_shuffle4(a, a, [3, 2, 1, 0])
 }
 
+/// Loads unaligned 64-bits of integer data from memory into new vector.
+///
+/// `mem_addr` does not need to be aligned on any particular boundary.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_loadu_si64)
+#[inline]
+#[target_feature(enable = "sse")]
+#[cfg_attr(test, assert_instr(movq))]
+#[stable(feature = "simd_x86_mm_loadu_si64", since = "1.46.0")]
+pub unsafe fn _mm_loadu_si64(mem_addr: *const u8) -> __m128i {
+    _mm_set_epi64x(ptr::read_unaligned(mem_addr as *const i64), 0)
```

Comment:
`_mm_set_epi64x` requires the `"sse2"` target feature, but this intrinsic only has `"sse"`. You need to use a different function to build the vector.

Reply:
The feature mismatch causes the test failure since the compiler can't inline `_mm_set_epi64x`.

Reply:
Ah, I didn't thought about this. I thought only those functions that are available are imported instead of having all functions imported, looks like a potential foot gun for new contributors. I wonder if there is sort of an import for directed acyclic graph with each file contain all the dependencies available. Like `sse2` depends on `sse` so `sse` should not import `sse2`.

But that looks like another issue, I will find out which function this could use.

Reply:
The build still failed, it doesn't seemed like that's the case. My guess is `movq` not being there.

Reply:
What seems to be happening now is that the 32-bit x86 calling convention is returning `__m128i` by reference, so LLVM is just entirely skipping the use of vector registers here. I would just disable the `assert_instr` on x86 with `#[cfg_attr(all(test, not(target = "x86")), assert_instr(movq))]`.

Reply:
Is that correct? I thought it needs to have `movq`? How do we add it back?

Reply:
It's fine. The intrinsics don't *have* to generate a particular instruction, it is acceptable for the compiler to generate code using different instructions that has the same effect.

Reply:
But isn't generating the original instruction that it should be faster rather than having multiple instructions to do the same thing?